apps/vscode/src/main.ts

```diff
@@ -125,9 +126,44 @@ export async function activate(context: vscode.ExtensionContext) {
 
   // activate providers common to browser/node
   activateCommon(context, host, engine, commands);
+
+  // if positron
+  setEditorOpener();
+
+  vscode.workspace.onDidChangeConfiguration(async (event) => {
+    if (event.affectsConfiguration('quarto.defaultEditor')) {
+      setEditorOpener();
+    }
+  });
```

Comment:
The disposable returned here needs to be captured so it can be disposed of at some point (like `documentOpenHandler`)